### ISSUE TITLE
Script Address Format

### ISSUE THREAD

**Issue by Abstrct** [litecoin-foundation/LoafWallet#22]
The script address format has changed from 3 addresses to M addresses in the latest released. This is preventing incompatibility with all systems that use script addresses.  

![img_3125](https://cloud.githubusercontent.com/assets/196776/25304620/19fa8662-2739-11e7-9af5-f1d1411c42bc.PNG)

You can easily see how the same addresses look different when compared in the block explorer and the same transaction within Loaf Wallet. 

![img_3126](https://cloud.githubusercontent.com/assets/196776/25304626/35920904-2739-11e7-93e9-8f925319ad38.PNG)

https://live.blockcypher.com/ltc/tx/3cb8647f18cd4a84c9784026bbce09feb8fd3cf02246fbde6c7d88100273b0f7/

Line in question appears to be #30 here:
https://github.com/litecoin-association/LoafWallet/blob/master/BreadWallet/NSString%2BBitcoin.h

```
#define BITCOIN_SCRIPT_ADDRESS      50
```

I believe it should be 

```
#define BITCOIN_SCRIPT_ADDRESS      5
```

Unless this is intentional, in which case, is there a push to get everything else to move to the new format? 

### Block Explorers

* http://explorer.litecoin.net/t/3JTeuNQFaq
* http://ltc.blockr.io/tx/info/3cb8647f18cd4a84c9784026bbce09feb8fd3cf02246fbde6c7d88100273b0f7
* https://live.blockcypher.com/ltc/tx/3cb8647f18cd4a84c9784026bbce09feb8fd3cf02246fbde6c7d88100273b0f7/
* https://chainz.cryptoid.info/ltc/tx.dws?3cb8647f18cd4a84c9784026bbce09feb8fd3cf02246fbde6c7d88100273b0f7.htm

### Wallets

* Litecoin Core
* Coinomi
* Exodus
* litevault.net

### Libraries

* pyCoin

### Other Services

* Coindroids

**Comment by losh11** [comment]
> Unless this is intentional, in which case, is there a push to get everything else to move to the new format?

The P2SH prefix is being changed from 3 (5) to M (50). Litecoin Core 0.13.3 release supports this by default. All wallets are recommended to update their P2SH prefix for mainnet and testnet. Block explorers are being updated.

**Comment by Abstrct** [comment]
Would it be at all possible to roll this change back for loafWallet and instead work on a more coordinated rollout? I'm happy to help with said coordination across services and platforms.

**Comment by Abstrct** [comment]
A bit of added context, I just started testing Litecoin support in Coindroids, a game that uses only script for all system addresses. Loafwallet especially was a contributing factor for choosing to use Litecoin as well, and I have been heavily suggesting that any players with an iOS device use Loafwallet.

With this change, now half my users (including me!) can't play from mobile.

Now, I can easily change the format we set our addresses to to mimic the change from 5 to 50, that part I'm not concerned about. The part that is very difficult for us is that the change is not uniform across all litecoin infrastructure. If I make that change then my users using other wallets like Coinomi (android) or Exodus, or litevault.net now can't play. It also heavily complicates support issues and debugging when all block explorers don't match the ins and outs properly either. 

In short, we are left choosing between one set of users or the other. 

If we can roll this loafwallet change back for now, and then work with each of these groups to get the change done across them all in a relatively short timeframe, this would prevent this entire situation. 

I completely understand that 3 (5) was an oversight to choose originally, as it means that script addresses for Litecoin and Bitcoin look exactly the same, but the timing on this is just really bad for Coindroids support. Over the last month since Coindroids started, it has accounted for 100% of transactions to a script address.

**Comment by losh11** [comment]
I'm going to push for the new P2SH prefix to be push across all Litecoin projects by the end of the week. Sorry about this though.

**Comment by Abstrct** [comment]
Going to push now, or had already started to push? Again, is there any chance at all we can roll back this change and work together on a coordinated rollout?

Is there a mailing this or something I'm not aware of where changes like this are discussed? I just want to make sure I'm better informed moving forward if I'm going to be working in the Litecoin space.

**Comment by losh11** [comment]
I might be able to add an option into settings allowing the user to choose what P2SH prefix to use. I'll try to add this to the v1.0.2 release. 

For Litecoin development discussion, it is best to use #litecoin-dev on IRC.

> already started to push?

Already started.

**Comment by Abstrct** [comment]
I don't think an option is necessary, as no users should ever want to change their address format - litecoin should only have one. Backwards compatibility when sending, with a warning, would be optimal, but even that seems like a lot of added code. 

The 3 (5) script format was in the codebase for 6 years, I'm just a bit concerned that the change will take over a week to propagate across projects.

**Comment by mperklin** [comment]
I've taken a look at how the litecoin project is handling the migration from 3*** to M*** addresses.

They define two separate version bytes for both old and new, and when validating the address they compare against both versions. If it matches _**either**_ then the address is valid.

This either/or comparison can be seen here: https://github.com/litecoin-project/litecoin/blob/master/src/base58.cpp#L221

Litecoin's functionality differs greatly from how the new version byte is implemented in LoafWallet. LoafWallet only validates new M*** addresses and claims that all 3*** addresses are invalid.

I recommend updating LoafWallet's address validation to match Litecoin's, so that scanning either a 3*** or M*** address will validate successfully.
